packages/devtools_app/lib/src/memory/memory_treemap.dart

```diff
@@ -0,0 +1,334 @@
+          String libraryName = theClass.libraryUri.toString();
+          if (libraryName.isEmpty) {
+            libraryName = libraryGroup;
+          }
+
+          // Map class names to familar user names.
+          final predefined =
+              predefinedClasses[LibraryClass(libraryName, className)];
+          if (predefined != null) {
+            className = predefined.prettyName;
+          }
+
+          final symbol = Symbol(
+            name: 'new $className',
+            size: shallowSize,
+            libraryUri: libraryName,
+            className: className,
+          );
+
+          Map<String, TreemapNode> currentChildren = rootChildren;
+          final TreemapNode parentReset = currentParent;
+          for (String pathPart in symbol.parts) {
+            currentChildren.putIfAbsent(
+              pathPart,
+              () {
+                final TreemapNode node = TreemapNode(
+                    name: pathPart, childrenMap: <String, TreemapNode>{});
+                currentParent.addChild(node);
+                return node;
+              },
+            );
+            currentChildren[pathPart].byteSize += symbol.size;
+            currentParent = currentChildren[pathPart];
+            currentChildren = currentChildren[pathPart].childrenMap;
+          }
+          currentParent = parentReset;
+        }
+      },
+    );
+
+    root.byteSize = root.childrenMap.values
+        .fold(0, (int current, TreemapNode node) => current + node.byteSize);
+
+    final snapshotGraph = controller.snapshots.last.snapshotGraph;
+    // Add the external heap to the heat map.
+    root.childrenMap.putIfAbsent('External Heap', () {
+      final TreemapNode node = TreemapNode(
+        name: 'External Heap',
+        childrenMap: <String, TreemapNode>{},
+      );
+      root.addChild(node);
+      node.byteSize = snapshotGraph.externalSize;
+      return node;
+    });
+
+    // Add the filtered libraries/classes to the heat map.
+    root.childrenMap.putIfAbsent('All Filtered Libraries', () {
+      final node = TreemapNode(
+        name: 'All Filtered Libraries',
+        childrenMap: <String, TreemapNode>{},
+      );
+      root.addChild(node);
+      node.byteSize = snapshotGraph.shallowSize - root.byteSize;
+      return node;
+    });
+
+    root.byteSize = snapshotGraph.shallowSize + snapshotGraph.externalSize;
+
+    return InstructionsSize(root);
+  }
+
+  final TreemapNode root;
+}
+
+class TreemapNode extends TreeNode<TreemapNode> {
+  TreemapNode({
+    @required this.name,
+    this.byteSize = 0,
+    this.childrenMap = const <String, TreemapNode>{},
+  })  : assert(name != null),
+        assert(byteSize != null),
+        assert(childrenMap != null);
+
+  final String name;
+  int byteSize;
+  final Map<String, TreemapNode> childrenMap;
+
+  void addSize(int byteSize) {
+    this.byteSize += byteSize;
+  }
+
+  TreemapNode getChildWithName(String name) {
```

Comment:
avoid `get` in method names. Rename to `childWithName`, or better yet, just access the map directly where this method is used (see next comment)